modules/rtm/R/bayestools.R

```diff
@@ -0,0 +1,168 @@
+        exceeds_vec <- gelman_vec[exceeds]
+        exceeds_char <- sprintf('%s: %.2f', names(exceeds_vec), exceeds_vec)
+        exceeds_str <- paste(exceeds_char, collapse = '; ')
+        message('The following parameters exceed threshold: ', exceeds_str)
+        return(FALSE)
+    } else {
+        return(TRUE)
+    }
+}
+
+#' Quick BayesianTools prior creator for PROSPECT model 
+#' 
+#' @param custom_prior List containing `param_name`, `distn`, `parama`, `paramb`, and `lower`
+#' @inheritParams prospect
+#' @export
+prospect_bt_prior <- function(version, custom_prior = list()) {
+    col_names <- c('param_name', 'distn', 'parama', 'paramb', 'lower')
+    prior_default_list <- list(N = list('N', 'norm', 1.4, 0.8, 1),
+                               Cab = list('Cab', 'lnorm', log(40), 0.9, 0),
+                               Car = list('Car', 'lnorm', log(10), 1.1, 0),
+                               Cbrown = list('Cbrown', 'lnorm', log(1), 1.1, 0),
+                               Cw = list('Cw', 'lnorm', log(0.01), 1, 0),
+                               Cm = list('Cm', 'lnorm', log(0.009), 1, 0),
+                               residual = list('residual', 'lnorm', log(0.001), 2.5, 0)
+                               )    
+    prior_list <- modifyList(prior_default_list, custom_prior)
+    prior_df_all <- do.call(rbind.data.frame, prior_list)
+    colnames(prior_df_all) <- col_names
+    default_params <- defparam(paste0('prospect_', tolower(version)))
+    use_names <- c(names(default_params), 'residual')
+    prior_df <- prior_df_all[prior_df_all[['param_name']] %in% use_names,]
+    prior <- PEcAn.assim.batch::pda.create.btprior(prior_df)
+    return(prior)
+}
+
+#' Perform Bayesian inversion using BayesianTools package
+#' 
+#' Use samplers from the BayesianTools package to fit models to data. Like 
+#' `invert.auto`, this will continue to run until convergence is achieved 
+#' (based on Gelman diagnostic) _and_ the result has enough samples (as 
+#' specified by the user; see Details).
+#' 
+#' @details `custom_settings` is a list of lists, containing the following:
+#'  * `common` -- BayesianTools settings common to both the initial and subsequent samples.
+#'  * `init` -- BayesianTools settings for just the first round of sampling. 
+#'  This is most common for the initial number of iterations, which is the 
+#'  minimum expected for convergence.
+#'  * `loop` -- BayesianTools settings for iterations inside the convergence 
+#'  checking `while` loop. This is most commonly for setting a smaller 
+#'  iteration count than in `init`.
+#'  * `other` -- Miscellaneous (non-BayesianTools) settings, including:
+#'      - `sampler` -- String describing which sampler to use. Default is `DEzs`
+#'      - `use_mpsrf` -- Use the multivariate PSRF to check convergence. 
+#'      Default is `FALSE` because it may be an excessively conservative 
+#'      diagnostic.
+#'      - `min_samp` -- Minimum number of samples after burnin before stopping.
```

Comment:
max lag (lag.max) isn't shown here.....is this because it wasn't meant as an exhaustive list of settings?